### ISSUE TITLE
[JENKINS-3092] Changing number of executors requires restart of Hudson

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/remoting#931]
<p>I'm almost sure that this worked in previous releases: changing the number of<br/>
executors on the master in Hudson via the UI needs a restart of Hudson to be<br/>
recognized.  </p>

<p>Usecase: we have several parallel database clone jobs running, that almost take<br/>
no processor time. If all executors are blocked by those jobs we would like to<br/>
increase the number of executors dynamically, allowing our compile job get running. </p>

<p>(If this is intended then a message "please restart Hudson" would improve the<br/>
usability.)</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3092" target="_blank">Changing number of executors requires restart of Hudson</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mindless">mindless</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: remoting
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2009-06-06T15:52:44+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-25
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I'm almost sure that this worked in previous releases: changing the number of
executors on the master in Hudson via the UI needs a restart of Hudson to be
recognized.  </p>

<p>Usecase: we have several parallel database clone jobs running, that almost take
no processor time. If all executors are blocked by those jobs we would like to
increase the number of executors dynamically, allowing our compile job get running. </p>

<p>(If this is intended then a message "please restart Hudson" would improve the
usability.)</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: PC, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3092?focusedId=128992&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128992">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3223 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3223 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3092?focusedId=128993&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128993">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>looks like it just needs an updateComputerList() call after saving the main
config.  you could even do this in your current version of hudson by entering
this in the script console after saving the config change:
hudson.model.Hudson.instance.updateComputerList()</p>
</pre>
</details>

</ul>
</details>

<p>looks like it just needs an updateComputerList() call after saving the main<br/>
config.  you could even do this in your current version of hudson by entering<br/>
this in the script console after saving the config change:<br/>
hudson.model.Hudson.instance.updateComputerList()</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3092?focusedId=128994&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128994">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : mindless
Path:
 trunk/hudson/main/core/src/main/java/hudson/model/Hudson.java
 trunk/www/changelog.html
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=18730" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=18730</a>
Log:
<span class="error">&#91;FIXED JENKINS-3092&#93;</span> Call updateComputerList() at end of doConfigSubmit so a change
in number of executors for master node takes effect immediately.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : mindless<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/Hudson.java<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=18730" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=18730</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-3092&#93;</span> Call updateComputerList() at end of doConfigSubmit so a change<br/>
in number of executors for master node takes effect immediately.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/remoting/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-3223%22">JENKINS-3223</a>]</i>
